edscrapers/transformers/base/helpers.py

```diff
@@ -84,9 +84,19 @@ def transform_keywords(tags_string):
     return keywords
 
 def extract_format_from_url(url):
-    path = urlparse(url).path
-    extension = path.split('.')[-1]
+    parsed_url = urlparse(url).geturl()
+    extension = parsed_url.split('.')[-1]
     if len(extension) < 5:
         return extension
     else:
-        return None
\ No newline at end of file
+        return None
+
+def extract_name_from_url(url):
+    parsed_url = urlparse(url).geturl()
+    file_name = parsed_url.split('/')[-1]
+    if len(file_name.split('.')) > 0:
```

Comment:
I think this is more reliable: `os.path.basename(urlparse(url).path)` - no splitting and hacking the list :) 

```
In [10]: os.path.basename(urlparse('https://ocrdata.ed.gov/downloads/projections/2013-14/Corporal-Punishment.xlsx').path)                                               
Out[10]: 'Corporal-Punishment.xlsx'
```

Reply:
Using ```urlparse(url).geturl()``` to parse filenames or formats in params/query/fragment.

Example:
https://ies.ed.gov/transfer.asp?location=peabody.vanderbilt.edu/x959.xml

If we use ```.path``` we will get ```transfer.asp```.

We still need a ```.split('.')``` to get a filename or format. Changed to ```.rsplit('.',1)``` to support multiple periods.